webapp/src/ts/modules/analytics/analytics-target-aggregates-sidebar-filter.component.ts

```diff
@@ -74,8 +85,17 @@ export class AnalyticsTargetAggregatesSidebarFilterComponent implements OnInit,
     }
   }
 
-  fetchAggregateTargets(facility) {
+  fetchAggregateTargetsByFacility(facility) {
     this.selectedFacility = facility;
-    this.facilitySelected.emit(this.selectedFacility);
+    this.facilitySelectionChanged.emit(this.selectedFacility);
   }
+
+  fetchAggregateTargetsByReportingPeriod() {
+    this.reportingPeriodSelectionChanged.emit(this.selectedReportingPeriod);
+  }
+}
+
+export enum ReportingPeriod {
+  CURRENT = 'targets.this_month.subtitle',
+  PREVIOUS = 'targets.last_month.subtitle'
```

Comment:
Usually, enums use simple words that don't change much and are easy to read. But labels (translation keys) don't seem a good fit for this, feels unexpected for enums. 

Is there a reason? Can we keep simple `current` and `previous` values? That would give `readonly reportingPeriods` (L20) more reasons to exist.